### DISCUSSION TITLE
Bruggeman transport efficiency submodel

### DISCUSSION THREAD

**Discussion by ch662** [pybamm-team/PyBaMM#2367] [Q&A] [answered]
Hello, I am trying to make a custom submodel from the Bruggeman submodel, but I can't seem to get the code to run. 
I noticed that the current github code for the Bruggeman submodel is entirely different than the code I get when I install pybamm 22.9 so I'm not sure what to make of that. 
I get the following error when I try to use the submodel , any ideas on how to get this to run or is there something I'm not seeing? 

I am running 
pybamm               22.9
Python 3.9.13

my code:

import pybamm
import numpy as np
import matplotlib.pyplot as plt

model = pybamm.lithium_ion.DFN(build = False)
    
model.submodels["electrolyte transport efficiency"] = pybamm.transport_efficiency.Bruggeman(
    model.param, "electrolyte"
)

model.build_model()

<img width="734" alt="Screenshot 2022-10-14 095801" src="https://user-images.githubusercontent.com/108564982/195891094-fab9f814-02ec-4485-9dd9-bbf99e3a6aa8.png">
<img width="803" alt="Screenshot 2022-10-14 095828" src="https://user-images.githubusercontent.com/108564982/195891093-2c0aad91-2aa5-4cf8-bbe9-3a380b844814.png">
<img width="758" alt="Screenshot 2022-10-14 100200" src="https://user-images.githubusercontent.com/108564982/195891090-28705eb4-f682-47e2-a997-9ba621a02004.png">

**Comment by valentinsulzer** [comment, marked as answer]
You need to capitalize "Electrolyte".
Also, `self._get_standard_transport_efficiency_variables` now takes a dictionary, keys should be domain strings (e.g. "negative electrode") and values should be the pybamm expression for that domain. This change is to allow more flexible model and domain definitions (half-cells, symmetric cells, dead lithium layer, etc), which is all work in progress
